src/store/store.ts

```diff
@@ -0,0 +1,28 @@
+import { create } from 'zustand';
+import type { Character } from '../app/home/page';
+
+export type StoreState = {
+  selectedItems: Character[];
+  addNewSelectedItems: (item: Character) => void;
+  removeSelectedItem: (id: number | string) => void;
+  removeAllSelectedItems: () => void;
+};
+
+export const useStore = create<StoreState>((set) => ({
+  selectedItems: [],
+  addNewSelectedItems: (item: Character) =>
+    set((state) => ({
+      selectedItems: state.selectedItems.some(
+        (i: Character) => i.id === item.id
+      )
+        ? state.selectedItems
+        : [...state.selectedItems, item],
+    })),
+  removeSelectedItem: (id) =>
+    set((state) => ({
+      selectedItems: state.selectedItems.filter(
+        (item: Character) => item.id !== id
+      ),
+    })),
```

Comment:
you can unite this actions in one `toggleItem` like 
```
  toggleItem: (item: Character) =>
    set((state) => {
      const isExist = state.selectedItems.some(
        (i: Character) => i.id === item.id
      );

      return {
        selectedItems: isExist
          ? state.selectedItems.filter(
              (character: Character) => character.id !== item.id
            )
          : [...state.selectedItems, item],
      };
    }),
```

and use in `src/app/home/components/CharacterItem/CharacterItem.tsx`